packages/lending/src/protocols/radiant-v2/lending-protocol.ts

```diff
@@ -0,0 +1,313 @@
+
+    const userBalancesMap: Record<
+      string,
+      { supplyBalance: string; variableBorrowBalance: string; usageAsCollateralEnabled: boolean }
+    > = {};
+    let j = 0;
+    for (let i = 0; i < this.reserves.length; i++) {
+      const { asset } = this.reserves[i];
+
+      const [aTokenBalance] = this.erc20Iface.decodeFunctionResult('balanceOf', returnData[j]);
+      j++;
+
+      const { currentVariableDebt, usageAsCollateralEnabled } = this.protocolDataProviderIface.decodeFunctionResult(
+        'getUserReserveData',
+        returnData[j]
+      );
+      j++;
+
+      userBalancesMap[asset.address] = {
+        supplyBalance: common.toBigUnit(aTokenBalance, asset.decimals),
+        variableBorrowBalance: common.toBigUnit(currentVariableDebt, asset.decimals),
+        usageAsCollateralEnabled,
+      };
+    }
+
+    return userBalancesMap;
+  }
+
+  async getPortfolios(account: string) {
+    const reserveDataMap = await this.getReserveDataMap();
+    const assetPriceMap = await this.getAssetPriceMap();
+    const userBalancesMap = await this.getUserBalancesMap(account);
+
+    const supplies: SupplyObject[] = [];
+    for (const token of tokensForDepositMap[this.chainId]) {
+      if (isWrappedNativeToken(this.chainId, token)) continue;
+
+      const reserveData = reserveDataMap[token.address];
+      const assetPrice = assetPriceMap[token.address];
+      const userBalance = userBalancesMap[token.address];
+
+      let usageAsCollateralEnabled = reserveData.usageAsCollateralEnabled;
+      if (Number(userBalance.supplyBalance) > 0) {
+        usageAsCollateralEnabled = userBalance.usageAsCollateralEnabled;
+      }
+
+      supplies.push({
+        token,
+        price: assetPrice,
+        balance: userBalance.supplyBalance,
+        apy: reserveData.supplyAPY,
+        usageAsCollateralEnabled,
+        ltv: reserveData.ltv,
+        liquidationThreshold: reserveData.liquidationThreshold,
+      });
+    }
+
+    const borrows: BorrowObject[] = [];
+    for (const token of tokensForBorrowMap[this.chainId]) {
+      if (isWrappedNativeToken(this.chainId, token)) continue;
+
+      const { variableBorrowAPY } = reserveDataMap[token.address];
+      const assetPrice = assetPriceMap[token.address];
+      const { variableBorrowBalance } = userBalancesMap[token.address];
+
+      borrows.push({
+        token,
+        price: assetPrice,
+        balances: [variableBorrowBalance],
+        apys: [variableBorrowAPY],
+      });
+    }
+
+    const portfolio = new Portfolio(this.chainId, this.id, this.market.id, supplies, borrows);
+
+    return [portfolio];
+  }
+
+  toUnderlyingToken(protocolToken: common.Token) {
+    return toToken(this.chainId, protocolToken);
+  }
+
+  toProtocolToken(underlyingToken: common.Token) {
+    return toRToken(this.chainId, underlyingToken);
+  }
+
+  isProtocolToken(token: common.Token) {
+    return isRToken(this.chainId, token);
+  }
+
+  async newSupplyLogic(params: SupplyParams) {
```

Comment:
I think that in the SDK, we should maintain the consistency of the interface, unless there is a particular scenario that requires otherwise. Therefore, perhaps we can continue to use `newSupplyLogic`.

Reply:
`LendingProtocol` implements the abstract `Protocol` to enable the adapter to utilize a unified interface for communication with each protocol. Therefore, the function name cannot be replaced with another name.